### PR TITLE
RavenDB-13180: Correctly escape characters in url on compare exchange…

### PR DESCRIPTION
**Pull request by garayx**
… operations.

### CODE REVIEW

File: test/SlowTests/Client/UniqueValues.cs [hunk 1]
```diff
@@ -295,5 +295,55 @@ public async Task CompareExchangeShouldBeRemovedFromStorageWhenDbGetsDeleted()
...
             }
         }
+
+        [Fact]
+        public async Task EnsureCanCompareExchangeWithCorrectCharsEscapeInKey()
+        {
+            DoNotReuseServer();
+            var store = GetDocumentStore();
+
+            const string cmpXchgKey1 = "emails/foo+test@email.com";
```
Comment (ml054) [inline test/SlowTests/Client/UniqueValues.cs]: can't we wrap this with a method and call it with arg and expected result to avoid code duplication?

Comment (garayx) [reply test/SlowTests/Client/UniqueValues.cs]: ofc I can, but I am not sure if this is necessary for test.

**Comment by JudahGabriel** [comment]
Egor, thanks for the fix! Can you confirm: if I store a cmpxchg with "emails/foo+test@ab.com", **will I be able to get that value using GetCompareExchangeValueOperation(keyWithPlusSign), or will I need to manually escape the keyWithPlusSign**?

**Comment by ml054** [comment]
@JudahGabriel according to test https://github.com/ravendb/ravendb/pull/8987/files#diff-d7adabe65d19221ae0f622e52c1b2751R307

you don't have to escape nothing.

**Comment by garayx** [comment]
> Egor, thanks for the fix! Can you confirm: if I store a cmpxchg with "emails/foo+test@ab.com", will I be able to get that value using GetCompareExchangeValueOperation(keyWithPlusSign), or will I need to manually escape the keyWithPlusSign?

Yes, the same key should be used for Put, Get, Delete operations despite 'special characters' used

**Review by arekpalinski — APPROVED** [review]

**Review by grisha-kotler — APPROVED** [review]

**Pull request merged**